goalie/go_mesh_seq.py

```diff
@@ -192,7 +242,9 @@ def indicate_errors(
         self._create_indicators()
 
         # Initialise adjoint solver generators on the MeshSeq and its enriched version
-        adj_sol_gen = self._solve_adjoint(**solver_kwargs)
+        adj_sol_gen = self._solve_adjoint(track_coefficients=False, **solver_kwargs)
+        # We need to track changes in the form coefficients if the problem is unsteady
+        solver_kwargs["track_coefficients"] = track_coefficients
```

Comment:
I don't like how this looks, so suggestions are welcome how to tidy it up. I had to allow passing `track_coefficients` through `solver_kwargs` only for the sake of existing tests in `adjoint/test_fp_iteration.py`.

We only want `track_coefficients=True` in the enriched mesh seq and only if the problem is unsteady. So I had to take `track_coefficients` out of `solver_kwargs` first and then put it back in... pretty ugly

Reply:
If parts of the source code exist purely for the sake of unit tests then this suggests to me that those unit tests could benefit from mocking. You can use mocking in a unit test to "fake" values of a class attribute or return values of a function/method, for example. As a downside, unit tests that use mocking do tend to look a bit ugly/hacky. However, that's better than having hacks in the source code.

Reply:
Thanks! Done in https://github.com/mesh-adaptation/goalie/pull/259/commits/1ce741b79639c1b34bf104202e5693586fc168ae